src/lib/ITensorMPS/src/mpo.jl

```diff
@@ -836,14 +836,15 @@ function ITensors.contract(
   N = length(A)
   N != length(B) &&
     throw(DimensionMismatch("lengths of MPOs A ($N) and B ($(length(B))) do not match"))
+  ResultType = typeof(B)
   # Special case for a single site
-  N == 1 && return MPO([A[1] * B[1]])
+  N == 1 && return ResultType([A[1] * B[1]])
   A = orthogonalize(A, 1)
   B = orthogonalize(B, 1)
   A = sim(linkinds, A)
   sA = siteinds(uniqueinds, A, B)
   sB = siteinds(uniqueinds, B, A)
-  C = MPO(N)
+  C = ResultType(N)
```

Comment:
```suggestion
  C = typeof(B)(N)
```
Just a style preference.